src/modules/api/user/user.controller.ts

```diff
@@ -1,7 +1,43 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { UserService } from './user.service';
+import { IResult } from 'src/common/interfaces/response';
+import { UserDTO } from './user.dto';
 
-@Controller('users/:id')
+@Controller('users')
 export class UserController {
-  constructor(private userService: UserService) {}
+  constructor(private readonly userService: UserService) {}
+
+  @Get(':id')
+  async getUserById(@Param('id') id) {
+    try {
+      const user = await this.userService.getUserById(id);
+      if (user) {
+        const userResp: UserDTO = {
+          name: user.name,
+          _id: user._id,
+          emailAddress: user.emailAddress,
+          joinedAt: user.joinedAt,
+          phoneNumber: user.phoneNumber,
+        };
+        const resp: IResult<UserDTO> = {
+          status: 'OK',
+          result: userResp,
+        };
+        return resp;
```

Comment:
How about returning them objects directly instead of declaring a var

Reply:
resp and errorResp u mean ?

Reply:
yes. 
```
return { status : 'OK', result: result} as IResult<UserDTO>;
```
in this way.